tests/test_callbacks.cpp

```diff
@@ -16,6 +16,8 @@
 
 int dummy_function(int i) { return i + 1; }
 
+static PyMethodDef def;
```

Comment:
Minor, but ... it's unfortunate that this very short and generic name is used only 220+ lines later.

But down there a longer name is annoying.

General: lumping chunks of unrelated code together is error prone (it's common practice in this code base, but that doesn't make it good).

Two birds with one stone: move the code below into a helper function up here? I'm thinking this line can be inside the helper function.